Orm/Xtensive.Orm.Tests/Linq/ContainsOverKeysCollectionTest.cs

```diff
@@ -44,15 +44,48 @@ public class MyEntity3 : Entity
     [Field]
     public MyEntity2 Entity { get; set; }
   }
+
+  [HierarchyRoot]
+  [KeyGenerator(KeyGeneratorKind.None)]
+  public class ComplexKeyEntity : Entity
+  {
+    [Field, Key(0)]
+    public int Id0 { get; private set; }
+
+    [Field, Key(1)]
+    public int Id1 { get; private set; }
+
+    [Field]
+    public string Name { get; set; }
+
+    public ComplexKeyEntity(int id0, int id1)
+      : base(id0, id1)
+    {
+    }
+  }
 }
 
 namespace Xtensive.Orm.Tests.Linq.ContainsOverKeysCollectionTest
 {
   [TestFixture]
   public class ContainsOverKeysCollectionTest : AutoBuildTest
   {
-    private IEnumerable<Key> keysList;
-    private IEnumerable<Key> keysArray; 
+    private IEnumerable<Key> keyListAsEnumerable;
+    private IEnumerable<Key> keyArrayAsEnumerable;
+    private ICollection<Key> keyListAsCollection;
+    private ICollection<Key> keyArrayAsCollection;
+    private List<Key> keyList;
+    private IList<Key> keyListAsInteface;

```

Comment:
What about `IReadOnlyCollection` and `IReadOnlyList` interfaces checks?

Reply:
I needed generic interfaces and types. I took some from the top of my mind. Taking more than one is just my habit. Since  ```typeof (IReadOnlyCollection<Key>).IsGenericType``` and ```typeof (IReadOnlyList<Key>).IsGenericType``` are both true it will work just fine.